pkg/streamingpromql/operators/aggregation.go

```diff
@@ -143,7 +143,6 @@ func (a *Aggregation) NextSeries(ctx context.Context) (types.InstantVectorSeries
 	// Iterate through inner series until the desired group is complete
```

Comment:
(wanted to leave this comment higher up on line 134 but GitHub won't let me)

What do you think about moving `start`, `end`, `interval` and `steps` to be values on the `Aggregation` struct, initialised in `SeriesMetadata`? Then we wouldn't have to compute them on every call to `NextSeries`, and we wouldn't have to pass them around in here as well.

Another option could be a `NewAggregation` constructor that sets them.